### PR TITLE
Refactoring of SFSyncTarget (second part)

### PR DESCRIPTION
**Pull request by wmathurin**
Data fetching now handled by SFSyncTarget sub classes
Next: allowing custom target

### CODE REVIEW

File: libs/SmartSync/SmartSync/Classes/Util/SFSyncTarget.m [hunk 1]
```diff
@@ -27,6 +27,11 @@
 #import "SFSoqlSyncTarget.h"
 #import "SFSoslSyncTarget.h"
 
+#define ABSTRACT_METHOD {\
```
Comment (khawkins) [inline libs/SmartSync/SmartSync/Classes/Util/SFSyncTarget.m]: Is this a concept we should generalize to the greater SDK (e.g. `SalesforceSDKConstants.h`)?

**Comment by khawkins** [comment]
Overall, looks great.

File: libs/SmartSync/SmartSync/Classes/Util/SFSyncTarget.m (hunk 1 above)
Comment (wmathurin) [inline libs/SmartSync/SmartSync/Classes/Util/SFSyncTarget.m]: Will do.

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**